### DISCUSSION TITLE
Error proxying conns

### DISCUSSION THREAD

**Discussion by calum-bird** [planetscale/discussion#36] [Bug Reports] [answered]
Hello,

I've been trying to use PlanetScale in a docker environment and upon execution of `pscale connect` with our database info, I get an error:

`error   couldn't retrieve TLS certificate for the client        {"error": "couldn't retrieve certs from cert source: Not Found"}`

I did some digging and it seems to stem from [sql-proxy](https://github.com/planetscale/sql-proxy/blob/c07c38144aa516d72235faf19b7d8480338d7bb7/proxy/client.go), despite never having installed sql-proxy myself. This error is followed by a message claiming a secure connection was established, but subsequent database requests cause another error:
`error   error proxying conns    {"error": "couldn't retrieve certs for instance: "INSTANCE INFO HERE": couldn't retrieve certs from cert source: Not Found"}`. The requests are never fulfilled.

I only encountered this error after deleting, then re-creating our database to remove demo data. The service tokens were updated as well, and I can use the cli without issue outside of a Docker container to send database requests using the shell. The installer I'm using in Docker is pscale_0.38.0_linux_amd64.deb.

Hope this is something easily fixed!

**Comment by aquarapid** [comment, marked as answer]
The `pscale` uses the `sql-proxy` code under the covers, so an error from there is not completely unexpected.

You can run `pscale connect` with the `--debug` flag, which prints some more debugging info, that might help narrow it down.  Armed with that info, I recommend you contact us at support@planetscale.com or via https://www.planetscale.com/support directly;  since it sounds like something specific to your database instance, and we will need to do some digging.

**Reply by calum-bird** [reply]
Sounds good, thanks for your reply!

**Reply by Anders-PlanetScale** [reply]
Our engineering team has added additional testing and better error reporting in place around the pscale connect proxy. Cert and instance validation is now more robust.

These changes were implemented with pscale release v0.45.0
https://github.com/planetscale/cli/releases/tag/v0.45.0

Please let us know if you continue to experience any issues with pscale connect
